### PR TITLE
simpler TinyJit inside TinyJit detection

### PR DESCRIPTION
**Pull request by chenyuxyz**
suggested in https://github.com/tinygrad/tinygrad/commit/73395b998b6bb5c7a5c4cabde72b29a81ab51ead#commitcomment-143660402

### CODE REVIEW

**Comment by geohot** [comment]
Oh interesting. We use capturing in some of the tests, maybe we forget to clear it.

**Comment by chenyuxyz** [comment]
The failure was due to capturing not clearing if an exception was raised (as expected) during capture step. I think capturing is closer to a Context than a list, and can be refactored this way

File: tinygrad/engine/jit.py [hunk 1]
```diff
@@ -146,20 +145,24 @@ def __call__(self, *args, **kwargs) -> ReturnType:
...
-        self.ret = self.fxn(*args, **kwargs)
-        if len(params:=get_parameters(self.ret)): Tensor.realize(params[0], *params[1:])
-        capturing.clear()
+        try:
+          self.ret = self.fxn(*args, **kwargs)
+          if len(params:=get_parameters(self.ret)): Tensor.realize(params[0], *params[1:])
+          capturing.clear()
+        except Exception as e:
+          capturing.clear()
+          raise e
```
Comment (geohot) [inline tinygrad/engine/jit.py:165]:
```
finally:
  capturing.clear()
```

**Comment by geohot** [comment]
Can we merge this? (and any reason to not switch to finally?)

**Comment by chenyuxyz** [comment]
yea let me update and merge, i missed your reply

**Pull request merged**